File: Sitana.Framework/CrashReporter/Data/Session.cs
using System;
using Ebatianos.DataTransfer;

namespace Ebatianos.CrashReporter.Data
{
	public class Session: IEcsStructure
	{
		public int? Id;
		public string ApplicationGuid;
		public string DeviceGuid;
		public DateAndTime Time;

        void IEcsStructure.Read(EcsReader reader)
		{
			Id = reader.HasField(0) ? (int?)reader.ReadInt32(0) : null;
			ApplicationGuid = reader.ReadString(1);
			DeviceGuid = reader.ReadString(2);
			Time = reader.ReadStructure<DateAndTime>(3);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			if (Id!=null) writer.Write(0, Id.Value);
			writer.Write(1, ApplicationGuid);
			writer.Write(2, DeviceGuid);
			writer.Write(3, Time);
		}
	}
}

File: OtherLanguages/PHP/CrashReporter/Data/Session.php
<?
class Session
{
	public $id;
	public $applicationGuid;
	public $deviceGuid;
	public $time;

	public function read($reader)
	{
		$this->id = $reader->hasField(0) ? $reader->readInteger(0) : null;
		$this->applicationGuid = $reader->readString(1);
		$this->deviceGuid = $reader->readString(2);
		$this->time = $reader->readStructure(3);
	}

	public function write($writer)
	{
		if ($this->id!=null) $writer->writeValue(0, $this->id);
		$writer->writeString(1, $this->applicationGuid);
		$writer->writeString(2, $this->deviceGuid);
		$writer->writeStructure(3, $this->time);
	}
}
?>